rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,126 @@
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+from .constants import CPT_LIDO, LIDO_STETH_SUBMITTED, STETH_MAX_ROUND_ERROR_WEI
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class LidoDecoder(DecoderInterface):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.steth_evm_address = A_STETH.resolve_to_evm_token().evm_address
+
+    def _decode_lido_staking_in_steth(self, context: DecoderContext, sender: ChecksumEvmAddress) -> DecodingOutput:  # noqa: E501
+        """Decode the submit of eth to lido contract for obtaining steth in return"""
+        # decode the ETH collateral amount submitted for staking in Lido
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        collateral_amount = token_normalized_value_decimals(
+            token_amount=amount_raw,
+            token_decimals=18,
+        )
+
+        # Searching for the already decoded event,
+        # containing the ETH transfer of the submit transaction
+        paired_event = None
+        action_from_event_type = None
+        for event in context.decoded_events:
+            if (
+                event.address == self.steth_evm_address and
+                event.asset == A_ETH and
+                event.balance.amount == collateral_amount and
+                event.event_type == HistoryEventType.SPEND and
+                event.location_label == sender
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Submit {collateral_amount} ETH to Lido for receiving stETH in exchange'  # noqa: E501
+                event.counterparty = CPT_LIDO
+                #  preparing next action to be processed when erc20 transfer will be decoded by rotki  # noqa: E501
+                #  needed because submit event is emitted prior to the erc20 transfer
+                paired_event = event
+                action_from_event_type = HistoryEventType.RECEIVE
+                action_to_event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                action_to_notes = f'Receive {{amount}} stETH in exchange of the deposited ETH'  # {amount} to be set by transform ActionItem  # noqa: E501,F541
```

Comment:
wording.

Also you don't need `f` sincethis is not an fstring. You are not using any variable in there. The lint warning `F541` seems correct.

```suggestion
                action_to_notes = 'Receive {amount} stETH in exchange for the deposited ETH'  # {amount} to be set by transform ActionItem  # noqa: E501
```